qiita_db/metadata_template/prep_template.py

```diff
@@ -850,13 +850,35 @@ def _get_predecessors(workflow, node):
         else:
             starting_job = None
             pt_artifact = self.artifact.artifact_type
-        workflows = [wk for wk in qdb.software.DefaultWorkflow.iter()
-                     if wk.artifact_type == pt_artifact and
-                     pt_dt in wk.data_type]
+
+        workflows = []
+        ST = qdb.metadata_template.sample_template.SampleTemplate
+        for wk in qdb.software.DefaultWorkflow.iter():
+            if wk.artifact_type == pt_artifact and pt_dt in wk.data_type:
+                wk_params = wk.parameters
+                reqs_satisfied = True
+
+                if wk_params['sample']:
+                    df = ST(self.study_id).to_dataframe(samples=list(self))
+                    for k, v in wk_params['sample'].items():
+                        if k not in df.columns or v not in df[k].unique():
+                            reqs_satisfied = False
+
+                if wk_params['prep']:
```

Comment:
Just to confirm, wk_params['prep'] doesn't need to be an elif instead of an independent if? I believe it's correct now but just to be sure.

Reply:
That's correct, there can be sample, prep or both kinds of parameters.